### ISSUE TITLE
gost test broken by KDF/PRF move to providers

### ISSUE THREAD

**Issue by mattcaswell** [openssl/openssl#9794]
Ever since #9662 was merged the GOST test has been failing:

````
        # INFO:  @ test/ssltestlib.c:964
        # SSL_connect() failed -1, 1
        # INFO:  @ test/ssltestlib.c:980
        # SSL_accept() failed -1, 1
        # ERROR: (bool) 'create_ssl_connection(serverssl, clientssl, SSL_ERROR_NONE) == true' failed @ test/gosttest.c:64
        # false
        # 00:60:90:A0:60:7F:00:00:error:3800007A:Provider routines::invalid digest:providers/common/kdfs/tls1_prf.c:174:
        # 00:60:90:A0:60:7F:00:00:error:14000044:SSL routines::internal error:ssl/t1_enc.c:81:
        not ok 2 - iteration 2
````

Note, that the gost test does not run by default. You must have the gost engine available and the environment variable `OPENSSL_GOST_ENGINE_SO` must be set to point to it.

**Comment by beldmit** [comment]
Should it be fixed in OpenSSL or in the engine?

**Comment by paulidale** [comment]
The error is failing to fetch the digest.

I've a pretty good idea this relates to the lack of engine support in the KDFs -- it's only using the provider fetch path to find things.  Once #9770 is merged, I plan on updating the KDFs to use the helper functions which do handle engines properly.

**Comment by paulidale** [comment]
I updated #9770 to include KDFs as well.

**Comment by levitte** [comment]
I can confirm.  I just ran `make test TEST=test_gost` on master, and then on a checkout of #9770.  The former fails, the latter works.

**Comment by paulidale** [comment]
Thanks, I'd not go that far yet.

I've merged #9770 which should fix this one.
